crypto/schnorr-adaptor-signature/sign_utils.go

```diff
@@ -9,168 +9,248 @@ import (
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 )
 
-const (
-	ModNScalarSize       = 32
-	FieldValSize         = 32
-	JacobianPointSize    = 33
-	AdaptorSignatureSize = JacobianPointSize + ModNScalarSize + 1
-)
-
+// encSign implements the core of the EncSign algorithm as defined in the spec.
+// It generates a 65-byte pre-signature using the given private key, nonce,
+// public key, message, and encryption key.
+//
+// The algorithm starts from step 9 since steps 1-8 are handled in EncSign
 func encSign(
 	privKey, nonce *btcec.ModNScalar,
 	pubKey *btcec.PublicKey,
 	m []byte,
 	t *btcec.JacobianPoint,
 ) (*AdaptorSignature, error) {
-	// R' = kG (with blinding in order to prevent timing side channel attacks)
+	// Step 9: Compute R = k0*G
+	// NOTE: later we will negate k0 if needed, and after that k = k0
 	k := *nonce
-	RHat, err := common.ScalarBaseMultWithBlinding(&k)
+	R, err := common.ScalarBaseMultWithBlinding(&k)
 	if err != nil {
 		return nil, fmt.Errorf("failed to compute kG: %w", err)
 	}
+	R.ToAffine()
+
+	// Step 10: Compute R + T
+	// Ensure T is in affine coordinates
+	tAffine := *t
+	tAffine.ToAffine()
 
-	// get R = R'+T
-	var R btcec.JacobianPoint
-	btcec.AddNonConst(RHat, t, &R)
-	// negate k and R if R.y is odd
-	affineRWithEvenY, needNegation := intoPointWithEvenY(&R)
-	R = *affineRWithEvenY
-	if needNegation {
+	var R0 btcec.JacobianPoint
+	btcec.AddNonConst(R, &tAffine, &R0)
+	R0.ToAffine()
```

Comment:
shouldn't we check R0 is point at infinity - https://github.com/ZhePang/Python_Specification_for_Schnorr_Adaptor/blob/main/reference.py#L182